src/ui/Templates/Template.ts

```diff
@@ -180,20 +180,19 @@ export class Template implements ITemplateProperties {
     return _.compact(allComponentsInsideCurrentTemplate);
   }
 
-  public async instantiateToElement(result: IQueryResult, templateOptions: IInstantiateTemplateOptions = {}): Promise<HTMLElement> {
+  public instantiateToElement(result: IQueryResult, templateOptions: IInstantiateTemplateOptions = {}): Promise<HTMLElement> | null {
     const mergedOptions = new DefaultInstantiateTemplateOptions().merge(templateOptions);
     const html = this.instantiateToString(result, mergedOptions);
 
     if (html == null) {
       return null;
     }
 
-    await this.ensureComponentsInHtmlStringHaveLoaded(html);
-
-    const template = this.buildTemplate(html, mergedOptions);
-    this.logger.trace('Instantiated result template', result, template);
-
-    return template;
+    return this.ensureComponentsInHtmlStringHaveLoaded(html).then(() => {
```

Comment:
I must have introduced this bug, but I don't understand how reverting to `promise` style from async/await solves the issue? Was the `async` keyword a problem?